lib/galaxy/model/__init__.py

```diff
@@ -2457,6 +2458,46 @@ def __init__(self, user, group):
         self.group = group
 
 
+class Notification(Base, Dictifiable, RepresentById):
+    __tablename__ = "notification"
+
+    id = Column(Integer, primary_key=True)
+    create_time = Column(DateTime, default=now)
+    update_time = Column(DateTime, default=now, onupdate=now)
+    publication_time = Column(DateTime, default=now)
+    expiration_time = Column(DateTime, default=now() + timedelta(days=30 * 6))
+    source = Column(String(32), index=True)
+    category = Column(String(64), index=True)
+    variant = Column(String(16), index=True)
+    content = Column(JSONType)
+
+    user_notification_associations = relationship("UserNotificationAssociation", back_populates="notification")
+
+    def __init__(self, source: str, category: str, variant: str, content):
+        self.source = source
+        self.category = category
+        self.variant = variant
+        self.content = content
+
+
+class UserNotificationAssociation(Base, RepresentById):
+    __tablename__ = "user_notification_association"
+
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey("galaxy_user.id"), index=True)
+    notification_id = Column(Integer, ForeignKey("notification.id"), index=True)
+    seen_time = Column(DateTime, nullable=True)
+    favorite = Column(Boolean, index=True, default=False)
+    deleted = Column(Boolean, index=True, default=False)
+
+    user = relationship("User", back_populates="all_notifications")
```

Comment:
You need to add the `all_notifications` relationship definition to the `User` class (same mirrored `back_populates` as well).